### DISCUSSION TITLE
Is it possible to reduce the generated input level to create or update mutations?

### DISCUSSION THREAD

**Discussion by infraymer** [MichalLytek/typegraphql-prisma#266] [Q&A]
Our Prisma scheme increases significantly in size. It has more than 150 models. The GraphQL schema is getting even bigger. We would like to reduce the generation of "input" nesting, so we use a maximum of 1 level of nesting for creating or updating models.

Is it possible to reduce the level of generation of input and output nesting?

I will be glad if you can tell me where to find this setting in the source code of the library. Thank you!

**Comment by MichalLytek** [comment]
There's no setting for reducing the nesting as Prisma itself uses nesting of the Prisma Client args.

You can only try to write your own arguments and inputs with all the fields flatten, then write resolvers that will translate this flat inputs into Prisma Client nested inputs call.

**Reply by infraymer** [reply]
Thanks! Could you share links to the files where you need to implement this approach?
